### PR TITLE
Add linux manywheel python 3.13 binary workflows

### PR DESCRIPTION
**Pull request by atalman**
Test with passing linux manywheel workflows is here: https://github.com/pytorch/pytorch/pull/121979
Builder PR already merged: https://github.com/pytorch/builder/pull/1910

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -19,4 +19,3 @@ ninja
 # setuptools was removed from default python install
 setuptools ; python_version >= "3.12"
 packaging
-optree>=0.11.0
```
Comment (Skylion007) [inline requirements.txt:22]: @XuehaiPan Probably need a new release with updated pybind11

Comment (XuehaiPan) [reply requirements.txt:22]: I plan to update the `pybind11` tag to `stable`.

Comment (XuehaiPan) [reply requirements.txt:22]: Updated.

- #130139

Comment (atalman) [reply requirements.txt:22]: hi @XuehaiPan was this version bump tested with py 3.13 ?

Comment (XuehaiPan) [reply requirements.txt:22]: @atalman Yes. All tests passed with Python 3.13 beta version.

<details>
<summary>Install log from SDist</summary>

```console
$ pyenv install 3.13.0b3 --debug
$ pyenv shell 3.13.0b3-debug
$ python3 --version              
Python 3.13.0b3
$ python3 -m pip download optree==0.12.0
$ python3 -m pip install -v ./optree-0.12.0.tar.gz 
Using pip 24.1.1 from /Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/lib/python3.13/site-packages/pip (python 3.13)
/Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/lib/python3.13/site-packages/pip/_internal/metadata/importlib/_envs.py:111: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  from pip._vendor.pkg_resources import find_distributions
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
Processing ./optree-0.12.0.tar.gz
  Running command pip subprocess to install build dependencies
  Using pip 24.1.1 from /Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/lib/python3.13/site-packages/pip (python 3.13)
  /Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/lib/python3.13/site-packages/pip/_internal/metadata/importlib/_envs.py:111: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    from pip._vendor.pkg_resources import find_distributions
  Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
  Collecting setuptools
    Using cached https://pypi.tuna.tsinghua.edu.cn/packages/42/54/2a8ecfcc9a714a6fbf86559a4b0f50b126a4ac4269ea8134f2c75c3e73de/setuptools-70.2.0-py3-none-any.whl (930 kB)
  /Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/lib/python3.13/site-packages/pip/_internal/metadata/importlib/_dists.py:73: DeprecationWarning: Unimplemented abstract methods {'locate_file'}
    return cls(files, info_location)
  Collecting pybind11
    Using cached https://pypi.tuna.tsinghua.edu.cn/packages/84/fb/1a249de406daf2b4ebd2d714b739e8519034617daec085e3833c1a3ed57c/pybind11-2.13.1-py3-none-any.whl (238 kB)
  Installing collected packages: setuptools, pybind11
  /Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/lib/python3.13/site-packages/pip/_internal/metadata/importlib/_dists.py:73: DeprecationWarning: Unimplemented abstract methods {'locate_file'}
    return cls(files, info_location)
    Creating /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-build-env-ymcerg89/overlay/bin
    changing mode of /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-build-env-ymcerg89/overlay/bin/pybind11-config to 755
  Successfully installed pybind11-2.13.1 setuptools-70.2.0
  Installing build dependencies ... done
  Running command Getting requirements to build wheel
  /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-build-env-ymcerg89/overlay/lib/python3.13/site-packages/setuptools/sandbox.py:15: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources
  running egg_info
  writing optree.egg-info/PKG-INFO
  writing dependency_links to optree.egg-info/dependency_links.txt
  writing requirements to optree.egg-info/requires.txt
  writing top-level names to optree.egg-info/top_level.txt
  reading manifest file 'optree.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  adding license file 'LICENSE'
  writing manifest file 'optree.egg-info/SOURCES.txt'
  Getting requirements to build wheel ... done
  Running command Preparing metadata (pyproject.toml)
  /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-build-env-ymcerg89/overlay/lib/python3.13/site-packages/setuptools/sandbox.py:15: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources
  running dist_info
  creating /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-modern-metadata-op4kt26b/optree.egg-info
  writing /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-modern-metadata-op4kt26b/optree.egg-info/PKG-INFO
  writing dependency_links to /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-modern-metadata-op4kt26b/optree.egg-info/dependency_links.txt
  writing requirements to /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-modern-metadata-op4kt26b/optree.egg-info/requires.txt
  writing top-level names to /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-modern-metadata-op4kt26b/optree.egg-info/top_level.txt
  writing manifest file '/private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-modern-metadata-op4kt26b/optree.egg-info/SOURCES.txt'
  reading manifest file '/private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-modern-metadata-op4kt26b/optree.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  adding license file 'LICENSE'
  writing manifest file '/private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-modern-metadata-op4kt26b/optree.egg-info/SOURCES.txt'
  creating '/private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-modern-metadata-op4kt26b/optree-0.12.0.dist-info'
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: typing-extensions>=4.5.0 in /Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/lib/python3.13/site-packages (from optree==0.12.0) (4.12.2)
Building wheels for collected packages: optree
  Running command Building wheel for optree (pyproject.toml)
  /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-build-env-ymcerg89/overlay/lib/python3.13/site-packages/setuptools/sandbox.py:15: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-14.5-arm64-cpython-313-pydebug
  creating build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree
  copying optree/accessor.py -> build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree
  copying optree/version.py -> build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree
  copying optree/functools.py -> build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree
  copying optree/registry.py -> build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree
  copying optree/__init__.py -> build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree
  copying optree/ops.py -> build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree
  copying optree/utils.py -> build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree
  copying optree/typing.py -> build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree
  creating build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/integration
  copying optree/integration/__init__.py -> build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/integration
  copying optree/integration/numpy.py -> build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/integration
  copying optree/integration/jax.py -> build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/integration
  copying optree/integration/torch.py -> build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/integration
  running egg_info
  writing optree.egg-info/PKG-INFO
  writing dependency_links to optree.egg-info/dependency_links.txt
  writing requirements to optree.egg-info/requires.txt
  writing top-level names to optree.egg-info/top_level.txt
  reading manifest file 'optree.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  adding license file 'LICENSE'
  writing manifest file 'optree.egg-info/SOURCES.txt'
  copying optree/_C.pyi -> build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree
  copying optree/py.typed -> build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree
  running build_ext
  /Users/PanXuehai/.pyenv/shims/cmake /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-req-build-8nnolds9 -DCMAKE_BUILD_TYPE=Release -DCMAKE_LIBRARY_OUTPUT_DIRECTORY_RELEASE=/private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-req-build-8nnolds9/build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree -DCMAKE_ARCHIVE_OUTPUT_DIRECTORY_RELEASE=/private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-req-build-8nnolds9/build/temp.macosx-14.5-arm64-cpython-313-pydebug -DPYTHON_EXECUTABLE=/Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/bin/python3 -DPYTHON_INCLUDE_DIR=/Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/include/python3.13d -DPYBIND11_CMAKE_DIR=/private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-build-env-ymcerg89/overlay/lib/python3.13/site-packages/pybind11/share/cmake/pybind11
  -- The CXX compiler identification is AppleClang 15.0.0.15000309
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/usr/bin/g++ - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Use Python version: 3.13.0
  -- Use Python executable: "/Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/bin/python3"
  -- Detected Python include directory: "/Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/include/python3.13d"
  -- Detected Pybind11 CMake directory: "/private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-build-env-ymcerg89/overlay/lib/python3.13/site-packages/pybind11/share/cmake/pybind11"
  CMake Warning (dev) at /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-build-env-ymcerg89/overlay/lib/python3.13/site-packages/pybind11/share/cmake/pybind11/FindPythonLibsNew.cmake:101 (message):
    Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
    are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
    the cmake_policy command to set the policy and suppress this warning, or
    preferably upgrade to using FindPython, either by calling it explicitly
    before pybind11, or by setting PYBIND11_FINDPYTHON ON before pybind11.
  Call Stack (most recent call first):
    /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-build-env-ymcerg89/overlay/lib/python3.13/site-packages/pybind11/share/cmake/pybind11/pybind11Tools.cmake:50 (find_package)
    /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-build-env-ymcerg89/overlay/lib/python3.13/site-packages/pybind11/share/cmake/pybind11/pybind11Common.cmake:202 (include)
    /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-build-env-ymcerg89/overlay/lib/python3.13/site-packages/pybind11/share/cmake/pybind11/pybind11Config.cmake:250 (include)
    CMakeLists.txt:193 (find_package)
  This warning is for project developers.  Use -Wno-dev to suppress it.

  -- Found PythonInterp: /Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/bin/python3 (found suitable version "3.13", minimum required is "3.13.0")
  -- Found PythonLibs: /Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/lib/libpython3.13d.dylib
  -- Performing Test HAS_FLTO_THIN
  -- Performing Test HAS_FLTO_THIN - Failed
  -- Performing Test HAS_FLTO
  -- Performing Test HAS_FLTO - Success
  -- Found pybind11: /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-build-env-ymcerg89/overlay/lib/python3.13/site-packages/pybind11/include (found version "2.13.1")
  -- Configuring done (0.6s)
  -- Generating done (0.0s)
  -- Build files have been written to: /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-req-build-8nnolds9/build/temp.macosx-14.5-arm64-cpython-313-pydebug
  /Users/PanXuehai/.pyenv/shims/cmake --build . --config Release --parallel --target _C --
  [ 37%] Building CXX object src/CMakeFiles/_C.dir/registry.cpp.o
  [ 37%] Building CXX object src/CMakeFiles/_C.dir/optree.cpp.o
  [ 50%] Building CXX object src/CMakeFiles/_C.dir/treespec/constructor.cpp.o
  [ 62%] Building CXX object src/CMakeFiles/_C.dir/treespec/treespec.cpp.o
  [ 75%] Building CXX object src/CMakeFiles/_C.dir/treespec/unflatten.cpp.o
  [ 75%] Building CXX object src/CMakeFiles/_C.dir/treespec/flatten.cpp.o
  [ 87%] Building CXX object src/CMakeFiles/_C.dir/treespec/traversal.cpp.o
  [100%] Linking CXX shared module /private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-req-build-8nnolds9/build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/_C.cpython-313d-darwin.so
  [100%] Built target _C
  installing to build/bdist.macosx-14.5-arm64/wheel
  running install
  running install_lib
  creating build/bdist.macosx-14.5-arm64
  creating build/bdist.macosx-14.5-arm64/wheel
  creating build/bdist.macosx-14.5-arm64/wheel/optree
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/accessor.py -> build/bdist.macosx-14.5-arm64/wheel/optree
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/_C.pyi -> build/bdist.macosx-14.5-arm64/wheel/optree
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/version.py -> build/bdist.macosx-14.5-arm64/wheel/optree
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/functools.py -> build/bdist.macosx-14.5-arm64/wheel/optree
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/registry.py -> build/bdist.macosx-14.5-arm64/wheel/optree
  creating build/bdist.macosx-14.5-arm64/wheel/optree/integration
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/integration/__init__.py -> build/bdist.macosx-14.5-arm64/wheel/optree/integration
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/integration/numpy.py -> build/bdist.macosx-14.5-arm64/wheel/optree/integration
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/integration/jax.py -> build/bdist.macosx-14.5-arm64/wheel/optree/integration
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/integration/torch.py -> build/bdist.macosx-14.5-arm64/wheel/optree/integration
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/__init__.py -> build/bdist.macosx-14.5-arm64/wheel/optree
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/ops.py -> build/bdist.macosx-14.5-arm64/wheel/optree
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/utils.py -> build/bdist.macosx-14.5-arm64/wheel/optree
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/_C.cpython-313d-darwin.so -> build/bdist.macosx-14.5-arm64/wheel/optree
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/py.typed -> build/bdist.macosx-14.5-arm64/wheel/optree
  copying build/lib.macosx-14.5-arm64-cpython-313-pydebug/optree/typing.py -> build/bdist.macosx-14.5-arm64/wheel/optree
  running install_egg_info
  Copying optree.egg-info to build/bdist.macosx-14.5-arm64/wheel/optree-0.12.0-py3.13.egg-info
  running install_scripts
  creating build/bdist.macosx-14.5-arm64/wheel/optree-0.12.0.dist-info/WHEEL
  creating '/private/var/folders/sq/7sf73d5s2qnb3w6jjsmhsw3h0000gn/T/pip-wheel-54zsxvkb/.tmp-6jhbm80d/optree-0.12.0-cp313-cp313d-macosx_14_0_arm64.whl' and adding 'build/bdist.macosx-14.5-arm64/wheel' to it
  adding 'optree/_C.cpython-313d-darwin.so'
  adding 'optree/_C.pyi'
  adding 'optree/__init__.py'
  adding 'optree/accessor.py'
  adding 'optree/functools.py'
  adding 'optree/ops.py'
  adding 'optree/py.typed'
  adding 'optree/registry.py'
  adding 'optree/typing.py'
  adding 'optree/utils.py'
  adding 'optree/version.py'
  adding 'optree/integration/__init__.py'
  adding 'optree/integration/jax.py'
  adding 'optree/integration/numpy.py'
  adding 'optree/integration/torch.py'
  adding 'optree-0.12.0.dist-info/LICENSE'
  adding 'optree-0.12.0.dist-info/METADATA'
  adding 'optree-0.12.0.dist-info/WHEEL'
  adding 'optree-0.12.0.dist-info/top_level.txt'
  adding 'optree-0.12.0.dist-info/RECORD'
  removing build/bdist.macosx-14.5-arm64/wheel
  Building wheel for optree (pyproject.toml) ... done
  Created wheel for optree: filename=optree-0.12.0-cp313-cp313d-macosx_14_0_arm64.whl size=300905 sha256=1f4857555449de862976a0e2ca17e32540806fa869405ff99688b14638d4c95a
  Stored in directory: /Users/PanXuehai/Library/Caches/pip/wheels/80/8b/d2/6bfdd69283cb0d8f468a43326731392a8b9f59b93876c0bcfd
/Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/lib/python3.13/site-packages/pip/_internal/metadata/importlib/_dists.py:73: DeprecationWarning: Unimplemented abstract methods {'locate_file'}
  return cls(files, info_location)
Successfully built optree
Installing collected packages: optree
  Attempting uninstall: optree
    Found existing installation: optree 0.12.0
    Uninstalling optree-0.12.0:
      Removing file or directory /Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/lib/python3.13/site-packages/optree-0.12.0.dist-info/
      Removing file or directory /Users/PanXuehai/.pyenv/versions/3.13.0b3-debug/lib/python3.13/site-packages/optree/
      Successfully uninstalled optree-0.12.0
Successfully installed optree-0.12.0
```

</details>

CI log for Python 3.13 pre-release: https://github.com/metaopt/optree/actions/runs/9811620123/job/27094201436?pr=137

Comment (albanD) [inline requirements.txt:22]: We can't remove it for every other versions, that would be not so good for testing, you can use `; python_version <= "3.12"` to preserve the current behavior for other versions.

Comment (albanD) [reply requirements.txt:22]: To allow us to move quickly, I would suggest disabling it for 3.13 in this PR and add it back in a follow up once we have a good version that is released.

**Review by albanD — APPROVED** [review]
SGTM!

**Pull request closed without merging**